ext/fiddle/pointer.c

```diff
@@ -261,7 +271,14 @@ rb_fiddle_ptr_s_malloc(int argc, VALUE argv[], VALUE klass)
     obj = rb_fiddle_ptr_malloc(s,f);
     if (wrap) RPTR_DATA(obj)->wrap[1] = wrap;
 
-    return obj;
+    if (rb_block_given_p()) {
+        if (f == NULL) {
```

Comment:
```suggestion
        if (!f) {
```

Because we don't use `== NULL` in other places.